astropy/coordinates/tests/test_intermediate_transformations.py

```diff
@@ -533,26 +519,24 @@ def test_teme_itrf():
 
 
 @pytest.mark.remote_data
-def test_earth_orientation_table():
+def test_earth_orientation_table(monkeypatch):
     """Check that we can set the IERS table used as Earth Reference.
 
     Use the here and now to be sure we get a difference.
     """
+    monkeypatch.setattr('astropy.utils.iers.conf.auto_download', True)
     t = Time.now()
     location = EarthLocation(lat=0*u.deg, lon=0*u.deg)
     altaz = AltAz(location=location, obstime=t)
     sc = SkyCoord(1*u.deg, 2*u.deg)
     # Default: uses IERS_Auto, which will give a prediction.
-    with catch_warnings() as w:
-        altaz_auto = sc.transform_to(altaz)
-
-    assert len(w) == 0
+    # Note: tests run with warnings turned into errors, so it is
+    # meaningful if this passes.
+    altaz_auto = sc.transform_to(altaz)
 
     with iers.earth_orientation_table.set(iers.IERS_B.open()):
-        with catch_warnings() as w:
+        with pytest.warns(AstropyWarning, match='after IERS data'):
```

Comment:
I must be going crazy. Pretty sure I needed some `.*` in the regex when I worked on it, but tests passed here without. 🤷